<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Ide</class>
 <widget class="QMainWindow" name="Ide">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>988</width>
    <height>762</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Processing IDE</string>
  </property>
  <property name="windowIcon">
   <iconset resource="icons/icons.qrc">
    <normaloff>:/processing_ide_app_icon.svg</normaloff>:/processing_ide_app_icon.svg</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="margin">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabShape">
       <enum>QTabWidget::Triangular</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="documentMode">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>architecture</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2" rowstretch="1,0,0">
        <property name="margin">
         <number>0</number>
        </property>
        <property name="spacing">
         <number>0</number>
        </property>
        <item row="2" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,0,0,0,0,0">
          <property name="spacing">
           <number>0</number>
          </property>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="mpZoomMinus">
            <property name="maximumSize">
             <size>
              <width>15</width>
              <height>15</height>
             </size>
            </property>
            <property name="text">
             <string>-</string>
            </property>
            <property name="autoRepeat">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="mpZoomSlider">
            <property name="minimum">
             <number>25</number>
            </property>
            <property name="maximum">
             <number>175</number>
            </property>
            <property name="value">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="invertedAppearance">
             <bool>false</bool>
            </property>
            <property name="invertedControls">
             <bool>false</bool>
            </property>
            <property name="tickPosition">
             <enum>QSlider::NoTicks</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="mpZoomPlus">
            <property name="maximumSize">
             <size>
              <width>15</width>
              <height>15</height>
             </size>
            </property>
            <property name="text">
             <string>+</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="mpZoomDisplay">
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>20</height>
             </size>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="mpResetZoom">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>50</width>
              <height>20</height>
             </size>
            </property>
            <property name="text">
             <string>100%</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="0">
         <widget class="cedar::proc::gui::View" name="mpProcessingDrawer"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>988</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="mpActionNew"/>
    <addaction name="separator"/>
    <addaction name="mpActionSave"/>
    <addaction name="mpActionSaveAs"/>
    <addaction name="mpActionLoad"/>
    <addaction name="mpRecentFiles"/>
    <addaction name="separator"/>
    <addaction name="mpActionExportSVG"/>
    <addaction name="separator"/>
    <addaction name="mpActionExit"/>
   </widget>
   <widget class="QMenu" name="menuPlugins">
    <property name="title">
     <string>Plugins</string>
    </property>
    <addaction name="mpActionManagePlugins"/>
    <addaction name="mpActionLoadPlugin"/>
   </widget>
   <widget class="QMenu" name="menuSimulation">
    <property name="title">
     <string>Simulation</string>
    </property>
    <addaction name="mpThreadsStartAll"/>
    <addaction name="mpThreadsStopAll"/>
    <addaction name="mpActionResetRootNetwork"/>
   </widget>
   <widget class="QMenu" name="mpMenuWindows">
    <property name="title">
     <string>Windows</string>
    </property>
    <addaction name="mpActionSettings"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="mpActionAbout"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="mpActionDuplicate"/>
    <addaction name="mpActionSelectAll"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <widget class="QMenu" name="menuTrigger_Connections">
     <property name="title">
      <string>Trigger Connections</string>
     </property>
     <addaction name="mpActionToggleTriggerVisibility"/>
     <addaction name="mpActionTriggerSmartHide"/>
    </widget>
    <addaction name="menuTrigger_Connections"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuPlugins"/>
   <addaction name="menuSimulation"/>
   <addaction name="mpMenuWindows"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="mpItemsWidget">
   <property name="features">
    <set>QDockWidget::DockWidgetFeatureMask</set>
   </property>
   <property name="windowTitle">
    <string>Elements</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="margin">
      <number>0</number>
     </property>
     <item>
      <widget class="QTabWidget" name="mpCategoryList"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="mpLogWidget">
   <property name="features">
    <set>QDockWidget::DockWidgetFeatureMask</set>
   </property>
   <property name="windowTitle">
    <string>Log</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="margin">
      <number>0</number>
     </property>
     <item>
      <widget class="cedar::aux::gui::Log" name="mpLog" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="mpToolsWidget">
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Tools</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="margin">
      <number>0</number>
     </property>
     <item>
      <widget class="QToolBox" name="mpToolBox">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="cedar::proc::gui::ArchitectureToolBox" name="mpArchitectureToolBox">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>124</width>
          <height>418</height>
         </rect>
        </property>
        <attribute name="label">
         <string>Architecture tools</string>
        </attribute>
       </widget>
       <widget class="QWidget" name="page_4">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>124</width>
          <height>418</height>
         </rect>
        </property>
        <attribute name="label">
         <string>Annotation Tools</string>
        </attribute>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="mpPropertiesWidget">
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Properties</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_5">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="margin">
      <number>0</number>
     </property>
     <item>
      <widget class="cedar::proc::gui::PropertyPane" name="mpPropertyTable"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="mpActionShowHideGrid"/>
   <addaction name="mpActionSave"/>
   <addaction name="mpActionResetRootNetwork"/>
   <addaction name="separator"/>
   <addaction name="mpActionToggleTriggerVisibility"/>
  </widget>
  <action name="mpActionNew">
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="mpActionSave">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="icons/icons.qrc">
     <normaloff>:/toolbaricons/save.svg</normaloff>:/toolbaricons/save.svg</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="mpActionSaveAs">
   <property name="text">
    <string>Save as ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="mpActionLoad">
   <property name="text">
    <string>Load ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="mpActionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="mpActionManagePlugins">
   <property name="text">
    <string>Manage ...</string>
   </property>
  </action>
  <action name="mpThreadsStartAll">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Start</string>
   </property>
  </action>
  <action name="mpThreadsStopAll">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
  </action>
  <action name="mpActionShowHideItems">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Steps</string>
   </property>
  </action>
  <action name="mpActionShowHideTools">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Tools</string>
   </property>
  </action>
  <action name="mpActionShowHideProperties">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Properties</string>
   </property>
  </action>
  <action name="mpActionShowHideLog">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Log</string>
   </property>
  </action>
  <action name="mpActionLoadPlugin">
   <property name="text">
    <string>Load plugin ...</string>
   </property>
  </action>
  <action name="mpActionShowHideGrid">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="icons/icons.qrc">
     <normaloff>:/toolbaricons/grid.svg</normaloff>:/toolbaricons/grid.svg</iconset>
   </property>
   <property name="text">
    <string>grid</string>
   </property>
   <property name="toolTip">
    <string>Toggle Grid</string>
   </property>
  </action>
  <action name="mpRecentFiles">
   <property name="text">
    <string>Recent files</string>
   </property>
  </action>
  <action name="mpActionSettings">
   <property name="text">
    <string>Settings...</string>
   </property>
  </action>
  <action name="mpActionAbout">
   <property name="text">
    <string>About...</string>
   </property>
  </action>
  <action name="mpActionResetRootNetwork">
   <property name="icon">
    <iconset resource="icons/icons.qrc">
     <normaloff>:/toolbaricons/reset.svg</normaloff>:/toolbaricons/reset.svg</iconset>
   </property>
   <property name="text">
    <string>Reset root network</string>
   </property>
   <property name="toolTip">
    <string>Resets the root network of the current architecture.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="mpActionExportSVG">
   <property name="text">
    <string>Export SVG ...</string>
   </property>
  </action>
  <action name="mpActionDuplicate">
   <property name="text">
    <string>Duplicate</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="mpActionSelectAll">
   <property name="text">
    <string>Select All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="mpActionToggleTriggerVisibility">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="icons/icons.qrc">
     <normaloff>:/toolbaricons/hide_triggers.svg</normaloff>
     <normalon>:/toolbaricons/show_triggers.svg</normalon>:/toolbaricons/hide_triggers.svg</iconset>
   </property>
   <property name="text">
    <string>toggle visibility</string>
   </property>
   <property name="toolTip">
    <string>Toggle visibility of triggers.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="mpActionTriggerSmartHide">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Smart hide</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>cedar::proc::gui::View</class>
   <extends>QGraphicsView</extends>
   <header location="global">cedar/processing/gui/View.h</header>
  </customwidget>
  <customwidget>
   <class>cedar::proc::gui::PropertyPane</class>
   <extends>QTableWidget</extends>
   <header>cedar/processing/gui/PropertyPane.h</header>
  </customwidget>
  <customwidget>
   <class>cedar::proc::gui::ArchitectureToolBox</class>
   <extends>QWidget</extends>
   <header>cedar/processing/gui/ArchitectureToolBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>cedar::aux::gui::Log</class>
   <extends>QWidget</extends>
   <header>cedar/auxiliaries/gui/Log.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons/icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>mpActionExit</sender>
   <signal>triggered()</signal>
   <receiver>Ide</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>493</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
